src/Configuration.cpp

```diff
@@ -24,6 +24,8 @@
 
 extern llvm::cl::opt<std::string> cl_transform;
 
+extern std::vector<std::string> cl_program_arguments;
```

Comment:
I would move the `llvm::cl::list<std::string> arguments` from `main()` here. That way, you don't need the intermediary `vector<string>`.

I.e. you'd have `static llvm::cl::list<std::string> cl_program_arguments(`... here and no changes to `main.cpp`.

Reply:
The reason I did it this way is that cl_program_arguments(...) gets the argument llvm::cl::Positional so cl_program_arguments(...) needs to be called after input_file(...). If we do as I describe in the next answer below this will not be an issue anymore.

Reply:
I think @carlleonardsson will have to give the final word, but I would just add a `std::string input_file` member to `Configuration` and have all command line option objects be declared in `Configuration.cpp`.

Reply:
ok, I think I found another solution that will also fix the "--" problem. The solution is to use the [cl::ConsumeAfter modifier](https://llvm.org/docs/CommandLine.html#id22). I will try that out tomorrow.